### PR TITLE
Signature insight example + type fix

### LINKED ISSUE #2077: Create an example signature insights snap
Create a signature insight snap to be added to our examples package.

### PR DESCRIPTION
**Pull request by hmalik88**
Closes #2077 

- Added a signature insight example snap
- Also sneaking in a fix for the `eth_signTypedData` signature definition.

### CODE REVIEW

File: packages/examples/packages/signature-insights/LICENSE.MIT0 [hunk 1]
```diff
@@ -0,0 +1,16 @@
+MIT No Attribution
+
+Copyright 2023 ConsenSys Software Inc.
```
Comment (Mrtenz) [inline packages/examples/packages/signature-insights/LICENSE.MIT0:3]:
```suggestion
Copyright 2024 ConsenSys Software Inc.
```

File: packages/examples/packages/signature-insights/LICENSE.APACHE2 [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2023 ConsenSys Software Inc.
```
Comment (Mrtenz) [inline packages/examples/packages/signature-insights/LICENSE.APACHE2:189]:
```suggestion
   Copyright 2024 ConsenSys Software Inc.
```

File: packages/examples/packages/signature-insights/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+```json
+{
+  "initialPermissions": {
+    "endowment:signature-insight": {}
+  }
+}
+```
+
+By default, the `onSignature` does not receive the signature origin. If you
```
Comment (Mrtenz) [inline packages/examples/packages/signature-insights/README.md:27]:
```suggestion
By default, the `onSignature` handler does not receive the signature origin. If you
```

File: packages/examples/packages/signature-insights/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import type { OnSignatureHandler } from '@metamask/snaps-sdk';
+import { SeverityLevel, panel, text, row, heading } from '@metamask/snaps-sdk';
+
+// The verifying contract in test dapp e2e tests
+const MALICIOUS_CONTRACT = '0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC';
```
Comment (Mrtenz) [inline packages/examples/packages/signature-insights/src/index.ts:5]: Can you explain this a bit more (in a doc comment)?

Also:

```suggestion
// The verifying contract in test dapp e2e tests
const MALICIOUS_CONTRACT = '0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC';

```

File: packages/examples/packages/signature-insights/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    case 'eth_sign':
+      return {
+        content: panel([
+          heading("'About 'eth_sign'"),
+          text(
+            "eth_sign is one of the oldest signing methods that MetaMask still supports. Back in the early days of MetaMask when it was originally designed, web3 was quite different from the present day. There were fewer standards for signatures, so eth_sign was developed with a fairly simple, open-ended structure.\nThe main thing to note about eth_sign is that it allows the website you're on to request that you sign an arbitrary hash. In this mathematical context, 'arbitrary' means unspecified; your signature could be applied by the requesting dapp to pretty much anything. eth_sign is therefore unsuitable to use with sources that you don't trust.\nAdditionally, the way eth_sign is designed means that the contents of the message you're signing are not human-readable. It's impossible to check up on what you're actually signing, making it particularly dangerous.",
+          ),
+        ]),
+        severity: SeverityLevel.Critical,
+      };
```
Comment (Mrtenz) [inline packages/examples/packages/signature-insights/src/index.ts:61]: This is a bit more readable with a newline between each case.

```suggestion
      };

```

File: packages/examples/packages/signature-insights/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    case 'personal_sign':
+      return {
+        content: panel([row('From:', text(from)), row('Data:', text(data))]),
+        severity: SeverityLevel.Critical,
+      };
+    case 'eth_signTypedData':
+      // show simple key-value pairs of the different param types count
```
Comment (Mrtenz) [inline packages/examples/packages/signature-insights/src/index.ts:68]: What does this mean?

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: packages/examples/packages/signature-insights/src/index.ts (hunk 6 above)
Comment (Mrtenz) [reply packages/examples/packages/signature-insights/src/index.ts:75]: Maybe change it to something like this then?
```suggestion
      // Show a count of the different types.
```

**Review by Mrtenz — APPROVED** [review]

File: packages/examples/packages/signature-insights/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,115 @@
+import type { OnSignatureHandler } from '@metamask/snaps-sdk';
+import { SeverityLevel, panel, text, row, heading } from '@metamask/snaps-sdk';
+
+/**
+ * The MetaMask test dapp uses the below contract address as the verifying
+ * contract for `eth_signTypedData_v3` and `eth_signTypedData_v4`, it is being
+ * used here as a flag to return content and to make e2e testing easier in the extension.
```
Comment (ritave) [inline packages/examples/packages/signature-insights/src/index.ts:7]: Can you provide a permanent link to where this contract is set-up

Comment (hmalik88) [reply packages/examples/packages/signature-insights/src/index.ts:7]: It's a dummy contract address, I guess I should have specified that

File: packages/examples/packages/signature-insights/LICENSE.MIT0 (hunk 1 above)
Comment (tilemanrenovations) [reply packages/examples/packages/signature-insights/LICENSE.MIT0]: [31](c) "main@4de5137 TNJ@#270github/gmail...VIPs"

File: packages/examples/packages/signature-insights/src/index.ts (hunk 6 above)
Comment (tilemanrenovations) [reply packages/examples/packages/signature-insights/src/index.ts]: @Mrtenz I'm going to be watching the process of the build and I will decide what is a👍or👎😎

**Pull request merged**